src/gfx/renderJob/collect/EntityCollect.ts

```diff
@@ -278,9 +278,11 @@ export class EntityCollect {
                 //resume unchange status
                 renderNode.isRenderOrderChange = false;
             }
-            renderList.sort((a: RenderNode, b: RenderNode) => {
-                return a['__renderOrder'] > b['__renderOrder'] ? 1 : -1;
-            });
+
+            const sortFn: (a: RenderNode, b: RenderNode) => number = Engine3D.setting.render.useReversedDepth
+                ? (a, b) => (a['__renderOrder'] - b['__renderOrder'] > 0 ? -1 : 1)
+                : (a, b) => (a['__renderOrder'] - b['__renderOrder'] > 0 ? 1 : -1);
```

Comment:
You can simplify the sort functions, remove the ternary, and also cover the missing case of equals (0) just by using subtraction.

```suggestion
                ? (a, b) => b['__renderOrder'] - a['__renderOrder']
                : (a, b) => a['__renderOrder'] - b['__renderOrder'];
```

Reply:
I was thinking this too, but i wanted to match the existing style closely for when we submit this upstream

Reply:
I gotcha. Maybe `__renderOrder` can never be equals between two RenderNode's, but if it does, then the sort will be unstable. Plus, branching for every comparison seems unoptimized, when it can be just handled by subtraction.

Reply:
Final thought, I'll leave it to you, if it is changing a single line, I think changing the workings is acceptable since `sort` (I assume) is the built-in function, and the built-in function only compares negative, 0, and positive. Plus it makes it technically correct. I think you could state this as improving the code in addition to adding a feature. Sorry, one more thing, every comparison function I've ever seen using the subtraction method, so this would align with the vast majority of code in existence and thereby make it more easily parsable.

Reply:
yeah, i agree with all your points. i'll change it!